### PR TITLE
Update Cloud SQL Postgres sample for Cloud Run

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: cloud-sql/postgres/Program.cs [hunk 1]
```diff
@@ -14,21 +14,55 @@
...
  */
 
+using System;
+using System.IO;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+using Google.Cloud.Diagnostics.AspNetCore;
+using CloudSql.Settings;
```
Comment (amanda-tarafa) [inline cloud-sql/postgres/Program.cs:24]: nit: sort alphabetically

File: cloud-sql/postgres/Program.cs [hunk 2]
```diff
@@ -14,21 +14,55 @@
...
 
-        public static IWebHost BuildWebHost(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
-                .UseStartup<Startup>()
+        public static IWebHostBuilder BuildWebHost(string[] args)
+        {
+            ReadAppSettings();
+
+            string port = Environment.GetEnvironmentVariable("PORT") ?? "8080";
+            string hostingUrl = String.Concat("http://0.0.0.0:", port);
```
Comment (amanda-tarafa) [inline cloud-sql/postgres/Program.cs:43]: nit: don't use String, better use string.
And also, the following suggestion is way more idiomatic:
```suggestion
            string hostingUrl = $"http://0.0.0.0:{port}");

```

File: cloud-sql/postgres/README.md [hunk 3]
```diff
@@ -56,6 +58,62 @@ with a PostgreSQL database when running in Google App Engine Flexible Environmen
...
+### Deploy to Cloud Run
+
+1.  Edit [appsettings.json](appsettings.json).  Update the connection string to be in the following
+    form:
+        
+        `Server=/cloudsql/your-project-id:us-central1:instance-name;Uid=aspnetuser;Pwd=;Database=votes`
+    
+    replacing `your-project-id` and `instance-name` with your Google Cloud Project's project id and 
+    the name of the Cloud SQL instance you created in a previous step. Also replace the values for
+    `Uid` and `Pwd` with the "User name" and "Password" you previously created on the Cloud SQL instance.
```
Comment (amanda-tarafa) [inline cloud-sql/postgres/README.md:70]: nit: Should they potentially replace the location as well?

File: cloud-sql/postgres/appsettings.json [hunk 4]
```diff
@@ -15,7 +15,14 @@
   "CloudSQL": {
     "Database": "PostgreSQL",
     "ConnectionString": "Uid=aspnetuser;Pwd=;Host=cloudsql;Database=votes"
+    // Uncomment and use the ConnectionString below when deploying to Cloud Run.
```
Comment (amanda-tarafa) [inline cloud-sql/postgres/appsettings.json:18]: I believe that comments are not allowed in JSON,  even if some parses do accetp it. And the readme seems to cover these already.

**Comment by amanda-tarafa** [comment]
Let me know if you want me to force merge. The CI failures are unrelated to this PR.

File: cloud-sql/postgres/Program.cs [hunk 5]
```diff
@@ -14,21 +14,67 @@
...
+
+    static class ProgramExtensions
+    {
+        // Google Cloud Run sets the PORT environment variable to tell this
+        // process which port to listen to.
+        public static IWebHostBuilder UsePortEnvironmentVariable(
+            this IWebHostBuilder builder)
+        {
+            string port = Environment.GetEnvironmentVariable("PORT");
+            if (!string.IsNullOrEmpty(port))
```
Comment (amanda-tarafa) [inline cloud-sql/postgres/Program.cs:73]: Same comment here as in the other one, not sure if there should be an 8080 fallback value for the port.

Comment (jsimonweb) [reply cloud-sql/postgres/Program.cs:73]: I attempted to update the command to be
string port = Environment.GetEnvironmentVariable("PORT") ?? "8080";

but encountered the same issue as noted in my comment on the Cloud SQL Postgres PR.
This suggested change causes the app to fail to run locally. 

As it currently is the app successfully works as expected running locally, on Cloud Run
and on App Engine.

Please force merge if everything else looks okay. Thanks!

Comment (amanda-tarafa) [reply cloud-sql/postgres/Program.cs:73]: OK, I've let the tests run now, they should be green, and you can merge on green. Thanks!!

**Comment by amanda-tarafa** [comment]
Force merging, failures are unrelated.

**Pull request merged**